server/src/main/java/com/project/yozmcafe/config/datasource/RoutingDataSource.java

```diff
@@ -0,0 +1,17 @@
+package com.project.yozmcafe.config.datasource;
+
+import org.springframework.jdbc.datasource.lookup.AbstractRoutingDataSource;
+import org.springframework.transaction.support.TransactionSynchronizationManager;
+
+public class RoutingDataSource extends AbstractRoutingDataSource {
+
+    public static final String MASTER_DATA_SOURCE = "master";
+    public static final String SLAVE_DATA_SOURCE = "slave";
+
+    @Override
+    protected Object determineCurrentLookupKey() {
+        boolean isReadOnly = TransactionSynchronizationManager.isCurrentTransactionReadOnly();
+
+        return isReadOnly ? SLAVE_DATA_SOURCE : MASTER_DATA_SOURCE;
```

Comment:
```suggestion
if(isCurrentTransactionReadOnly()){
    return SLAVE_DATA_SOURCE;
}

return MASTER_DATA_SOURCE;
```
static import와 함께 이건 어떤가요...?

Reply:
ㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋ ㅠㅠ 수정했슴다